src/software/firmware/srcs/battery.cpp

```diff
@@ -0,0 +1,82 @@
+/*
+    Copyright (C) 2020 Makers For Life
+*/
+/******************************************************************************
+ * @author Makers For Life
+ * @file battery.cpp
+ * @brief Battery related functions
+ *****************************************************************************/
+
+#pragma once
+
+// INCLUDES ===================================================================
+
+// Associated header
+#include "../includes/battery.h"
+
+// Externals
+#include "Arduino.h"
+
+// Internal
+#include "../includes/parameters.h"
+
+// PROGRAM =====================================================================
+
+uint16_t DEFAULT_BATTERY_VOLTAGE = 20;  // 20 volts;
+uint16_t DEFAULT_POWER_VOLTAGE = 24;    // 24 volts;
+uint16_t BATTERY_MAX_SAMPLES = 20;      // Maximum battery samples
+uint16_t BATTERY_SAMPLE[20];            // Array to store battery voltage samples
+uint16_t BATTERY_CURRENT_SAMPLE = 0;    // Current battery sample index
+uint16_t BATTERY_TOTAL_SAMPLES = 0;     // Battery total samples
```

Comment:
Not sure what is the risk of overflow with a uint16_t here (with a default raw value at 1024, I think we would overflow with 66 samples). What about using a uint32_t to be sure? (and maybe for all use uint32_t for all values)